internal/stack/_static/logstash_startup.sh

```diff
@@ -0,0 +1,38 @@
+#!/bin/bash
+
+set -euo pipefail
+
+LOGSTASH_HOME="/usr/share/logstash/"
+
+# logstash expects the key in pkcs8 format.
+# Hence converting the key.pem to pkcs8 format using openssl.
+create_cert() {
+  ls_cert_path="$LOGSTASH_HOME/config/certs"
+  openssl pkcs8 -inform PEM -in "$ls_cert_path/key.pem" -topk8 -nocrypt -outform PEM -out "$ls_cert_path/logstash.pkcs8.key"
```

Comment:
`$LOGSTASH_HOME/config/certs` being a path that is mounted container is unable to write to this location. I changed it in https://github.com/elastic/elastic-package/pull/1669/files#diff-2240cd26d5cca1ec7a1221db84f8da74fe89c916c26a4b99a073607319120bf0R176

Reply:
Hm, the `$LOGSTASH_HOME/config/certs` path is writable and `logstash.pkcs8.key` is being generated under it. Also, you used it in serverless mode. I am not enabling SSL between agent and Logstash, so will not affect, serverless stay same AS-IS. I think better to isolate SSL change with this PR.

```
$ ls config/certs
ca-cert.pem  cert.pem  elasticsearch.pem  key.pem  logstash.pkcs8.key
```

<img width="775" alt="image" src="https://github.com/elastic/elastic-package/assets/99575341/995b967c-0b2b-44be-9dc6-46f740c0df0a">

Reply:
It works locally. But on [buildkite](https://buildkiteartifacts.com/e0f3970e-3a75-4621-919f-e6c773e2bb12/01860528-c8cd-4386-b9ad-d0ff1f634a5d/018da56f-aa46-47b6-a535-bf1ba80d0d20/018da576-ac60-442c-92e5-2ecebcba7b89/build/elastic-stack-dump/check-with-logstash/logs/logstash.log?response-content-type=text%2Fplain&X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=ASIAQPCP3C7L2WZSQ533%2F20240214%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20240214T075008Z&X-Amz-Expires=600&X-Amz-Security-Token=IQoJb3JpZ2luX2VjEK3%2F%2F%2F%2F%2F%2F%2F%2F%2F%2FwEaCXVzLWVhc3QtMSJIMEYCIQDU8oN1%2F52FRMPFuWDdhQAKutx3tgDW4WXKY3RDJqyMGQIhANbHK5hcmxvCD5h3Jv0JKkhzyXCC0jkYEReWLRxCNriaKvoDCIX%2F%2F%2F%2F%2F%2F%2F%2F%2F%2FwEQABoMMDMyMzc5NzA1MzAzIgxHP%2FYLiK7Z2pfU%2BR0qzgOU1PGs%2Fj2QuygYhB5tyCGSu6DtMRboHaYN8ciu4ZG3QN4dDbad7oY1KfgE0vjwTcHMaptv7zpg7oFZ0jS2etvNR8vKEJlgz4FhXBcudxpQakuMEGF3KyPTSLw9aw1IxIzBrqtdo416u7%2B8tZ%2BYhraRoHdb6QUGgUUoVeDw9gnFHSL6hZt6Tn8Fag12cXFv0YIytOYkqqIcnGXpX1iLx7siPOmelqFKDcK4AQyMIve75tmoD1zDkLq1zAcb5hsjJLyFYKmXEj6HntUmxMuSPcfaunF1IutL9BNpZaBFcBQH6lFFoyofBKYyTGoZYiABq7AR%2Fqb%2BAPlQ%2BuHZj3U9o1yrDBY8pqIUXWUcHirLGrJ9X4l43rMlT0a4GzWOFE0Yia28sTfMJNVS4wLao0GwXgkXnFeUWC%2BtEawgUxK79KFAsUts3nTkIpkmpU0%2BEuzcZ7Bsttd3vH65ZwnpL%2FPxQKyC8B7ehiFPQ%2F5VFG%2FmKBzOZ80P8m3wPxtNZ%2FcTNsPuCKdB8jYwMI7WQPAoi7q31GjAwmjKs%2FCO8Djdh%2FchlGvcAy%2FcmX1YpkRRycQUNoJ0ibRsEHE0EKJwBepoBZd8M5%2FRCbS3mJ037zBCcKQEn%2BAwzfmwrgY6pAE6p4dvWXmykfQGGAM8SF3NTzeeDfCQ6iGipalf0cW%2FmIkqiP5czHncp9lCtlaO1XbHwcADsFohUgMIuX5WOVk12laJU%2Bg0scyojivVZs9btAJYoCvBdz9aa2bfJLB0PEXAtfA5pWfxUl%2BN2MIiCv%2BeYzIzEAdCgSDCoztNp2O514Oiwcu3nImzQpQ50UJtTEHsIq%2Fm%2FfAu%2BR%2FR0dQOgATeGTElkg%3D%3D&X-Amz-SignedHeaders=host&X-Amz-Signature=033bdfd18f667887b216501457125176cfdf68851dee0c68b26e40fe7bd0b047) it is an issue.

Reply:
Yeah, we should not write to a mounted directory.

Reply:
Yeah, this is I got to know yesterday, will change soon.